### ISSUE TITLE
Implement Getters and setters for objects

### ISSUE THREAD

**Issue by jasonwilliams** [boa-dev/boa#753]
We support getters and setters in our property descriptors, but we don't do anything with them.
In this example here: https://github.com/boa-dev/boa/blob/master/boa/src/object/internal_methods.rs#L102-L104 we check if a property exists, if it doesn't we should fallback to the getter and call that, but as you can see right now we just return undefined.

There will also need to be some syntax updates to support the get and set inside of objects, im not sure

Spec: https://tc39.es/ecma262/#sec-ordinary-object-internal-methods-and-internal-slots-get-p-receiver
MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get

This code should now work and give the expected result:
```javascript
let a = {
    get b() {
        return "test";
    }
}

a.b // "test"
```

## Where to start

Running the above right now would give an error and land here:
https://github.com/boa-dev/boa/blob/master/boa/src/exec/object/mod.rs#L29

`MethodDefinitionKind::Get` would need to be implemented

**Comment by HalidOdat** [comment]
I will be working on the execution part of getters and setters (after some needed refactoring) and the other parsing half is free for anyone to take.

**Comment by HalidOdat** [comment]
This issue is blocked on:
 - [ ] Object internal methods should live in GcObject (#835) 
 - [ ] Object internal methods should take a `Context` and return `Result<T>` (todo)

**Comment by Razican** [comment]
I believe this is implemented, right?

/cc @HalidOdat @tofpie

**Comment by tofpie** [comment]
As far as I can tell #986 is more or less a duplicate of this one, and it has been solved by #987.

**Comment by Razican** [comment]
> As far as I can tell #986 is more or less a duplicate of this one, and it has been solved by #987.

Perfect, let's close this :)
